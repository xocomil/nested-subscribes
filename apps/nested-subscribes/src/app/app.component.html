<div class="container">

  <div class="child">
    <nested-nested-subscribe></nested-nested-subscribe>
  </div>
  <div class="child">
    <nested-switch-map-observables></nested-switch-map-observables>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
  <div class="child">
    <nested-pokemon-list></nested-pokemon-list>
  </div>
</div>
