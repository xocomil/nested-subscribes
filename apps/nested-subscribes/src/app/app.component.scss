$height: calc(50% - 17.5px);

.container {
  display: flex;
  flex-wrap: wrap;
  column-gap: 10px;
  row-gap: 15px;

  margin: 10px;

  height: calc(100% - 20px);
}

.child {
  min-width: calc(50% - 13px);
  flex-grow: 1;
  min-height: $height;
  max-height: $height;
  overflow-y: auto;
  padding: 4px;
}
