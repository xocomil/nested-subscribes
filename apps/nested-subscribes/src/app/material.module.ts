import { A11yModule } from '@angular/cdk/a11y';
import { BidiModule } from '@angular/cdk/bidi';
import { DragDropModule } from '@angular/cdk/drag-drop';
import { ObserversModule } from '@angular/cdk/observers';
import { OverlayModule } from '@angular/cdk/overlay';
import { PlatformModule } from '@angular/cdk/platform';
import { PortalModule } from '@angular/cdk/portal';
import { CdkStepperModule } from '@angular/cdk/stepper';
import { CdkTableModule } from '@angular/cdk/table';
import { CdkTreeModule } from '@angular/cdk/tree';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatInputModule } from '@angular/material/input';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatTableModule } from '@angular/material/table';

@NgModule({
  exports: [
    // CDK
    A11yModule,
    BidiModule,
    ObserversModule,
    OverlayModule,
    PlatformModule,
    PortalModule,
    CdkStepperModule,
    CdkTableModule,
    CdkTreeModule,
    DragDropModule,
    // Material
    MatButtonModule,
    MatInputModule,
    MatProgressSpinnerModule,
    MatTableModule,
  ],
})
export class MaterialModule {}
