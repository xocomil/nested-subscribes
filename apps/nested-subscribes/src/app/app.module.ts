import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppComponent } from './app.component';
import { PokemonListComponent } from './components/pokemon-list/pokemon-list.component';
import { MaterialModule } from './material.module';

@NgModule({
  declarations: [AppComponent, PokemonListComponent],
  imports: [BrowserModule, BrowserAnimationsModule, MaterialModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
