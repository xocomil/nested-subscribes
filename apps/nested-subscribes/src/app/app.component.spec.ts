import { createComponentFactory } from '@ngneat/spectator/jest';
import { AppComponent } from './app.component';

const createComponent = createComponentFactory({ component: AppComponent });

describe('AppComponent', () => {
  it('should create the app', () => {
    const spectator = createComponent();

    expect(spectator).toBeTruthy();
  });

  it(`should have as title 'nested-subscribes'`, () => {
    const spectator = createComponent();

    expect(spectator.component.title).toEqual('nested-subscribes');
  });
});
