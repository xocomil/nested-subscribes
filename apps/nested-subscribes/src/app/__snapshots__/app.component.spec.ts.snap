// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<nested-subscribes-root
  __ngContext__={[Function LRootView]}
  title={[Function String]}
>
  <div
    class="container"
  >
    <div
      class="child"
    >
      <nested-nested-subscribe />
    </div>
    <div
      class="child"
    >
      <nested-switch-map-observables />
    </div>
    <div
      class="child"
    >
      <nested-one-observable />
    </div>
    <div
      class="child"
    >
      <nested-merge-map-sorted />
    </div>
  </div>
</nested-subscribes-root>
`;
